File: project-manager/src/main/java/uni/lodz/pl/projectmanager/task/model/Task.java
package uni.lodz.pl.projectmanager.task.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.SneakyThrows;
import org.apache.commons.lang3.StringUtils;
import org.springframework.web.bind.MissingRequestValueException;
import uni.lodz.pl.projectmanager.project.model.Project;
import uni.lodz.pl.projectmanager.sprint.model.Sprint;
import uni.lodz.pl.projectmanager.user.model.User;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@NoArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    private Long id;
    @Column(nullable = false)
    private String name;
    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;
    @Column(nullable = false)
    private LocalDateTime created;
    @ManyToOne(optional = false)
    private User createdBy;
    private LocalDate end;
    @ManyToOne
    private User assignedTo;
    @Column(nullable = false)
    private TaskState taskState;
    @ManyToOne(optional = false)
    private Project project;
    @ManyToOne
    private Sprint sprint;

    @SneakyThrows
    public Task(AddTaskDto taskDto, User author, User assingedTo, Project project, Sprint sprint) {
        if (StringUtils.isBlank(taskDto.getName())) throw new MissingRequestValueException("Name missing");
        if (StringUtils.isBlank(taskDto.getDescription()))
            throw new MissingRequestValueException("Description missing");
        if (project == null) throw new MissingRequestValueException("Project missing");
        LocalDate end = taskDto.getEnd() != null ? taskDto.getEnd() : null;
        this.sprint = sprint;
        this.name = taskDto.getName();
        this.description = taskDto.getDescription();
        this.created = LocalDateTime.now();
        this.createdBy = author;
        this.end = end;
        this.assignedTo = assingedTo;
        this.taskState = TaskState.TODO;
        this.project = project;
    }
}

File: project-manager-ui/src/model/task/Task.ts
import {Project} from "../project/Project";
import {User} from "../User";
import {TaskDto} from "./TaskDto";
import {TaskState} from "./TaskState";
import {Sprint} from "../sprint/Sprint";

export class Task {
  id: number;
  name: string;
  description: string;
  created: Date | null;
  createdBy: User;
  end: Date | null;
  assignedTo?: User;
  taskState: TaskState;
  project: Project;
  sprint?: Sprint;

  constructor(taskDto: TaskDto) {
    this.id = taskDto.id;
    this.name = taskDto.name;
    this.description = taskDto.description;
    const c = taskDto.created;
    this.created = c !== null ? new Date(c[0], c[1], c[2], c[3], c[4], c[5]) : null;
    this.createdBy = taskDto.createdBy;
    const e = taskDto.end;
    this.end = e !== null ? new Date(e[0], e[1], e[2]) : null;
    this.assignedTo = taskDto.assignedTo;
    this.taskState = taskDto.taskState;
    this.project = taskDto.project;
    this.sprint = taskDto.sprint;
    return;
  }
}
